File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        index1 = 0
        index2 = len(numbers) - 1
        res = []
        while True:
            curr = numbers[index1] + numbers[index2]
            if curr > target:
                index2 = index2 - 1
            elif curr < target:
                index1 = index1 + 1
            else:
                break

        res.append(index1 + 1)
        res.append(index2 + 1)
        return res

File: Data Structures & Algorithms/two-integer-sum/submission-2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> storage;
        int index1, index2;
        for (int i = 0; i < nums.size(); i++) {
            storage[target - nums[i]] = i;
        }
        for (int i = 0; i < nums.size(); i++) {
                if (storage.count(nums[i]) && i != storage[nums[i]]) {
                    index1 = i;
                    index2 = storage[nums[i]];
                    break;
                }
        }
        vector<int> res;
        res.emplace_back(index1);
        res.emplace_back(index2);
        return res;
    }
};
